src/library.js

```diff
@@ -2655,6 +2655,9 @@ LibraryManager.library = {
     var me = _emscripten_run_script_string;
     var len = lengthBytesUTF8(s);
     if (!me.bufferSize || me.bufferSize < len+1) {
+      if (!me.bufferSize) {
+        _atexit(function() { _free(me.buffer); });
+      }
```

Comment:
As in the comment before, I don't think this is worth it. It's fine to leak this, and it's less code and simpler to do so.

The only part I'm not sure of is how to call `emscripten_builtin_malloc` here, but maybe we already have a helper for that?